python/ray/train/data_parallel_trainer.py

```diff
@@ -468,7 +468,12 @@ def clear_lazy_checkpoint_marker():
                 logger.debug(
                     f"Deleting the stale lazy checkpoint marker file: {marker_file}."
                 )
-                marker_file.unlink()
+                # Multiple workers on the same node may delete this file at the
+                # same time. Return if the marker file has been deleted.
+                try:
+                    marker_file.unlink()
```

Comment:
Oh oops, I just opened this PR since I saw a flaky test caused by this 😅  https://github.com/ray-project/ray/pull/36426

I think I prefer just having the local rank 0 worker call unlink, but also fine with this.